Docs as code & doc ops updates (#1528)
writethedocs/www: docs/guide/doc-ops.rst

DocOps
======

author: Jodie Putrino & the Write the Docs community

What is DocOps, anyway?
-----------------------

Along with the advent of :doc:`/guide/docs-as-code`, the technical writing community 
has seen the growth of a new specialization called "DocOps" (or "doc-ops").

But what does DocOps actually mean, and who actually "does" DocOps?

> This subject came up for discussion during the WTD NA 2021 Unconference session. 

Reviewer: Maybe spell out "Write the Docs" for first mention (even here?), and even I don't know what NA is. Isn't it always called WTD Portland?

> This subject came up for discussion during the WTD NA 2021 Unconference session. 

Author: ```suggestion
This subject came up for discussion during an Unconference session at the Write the Docs North America 2021 conference in Portland. 
```

This page represents some of the ideas presented during that session.

Generally speaking, "DocOps" is like `DevOps`_. Instead of applying broadly to 
software development, though, DocOps specifically applies to the creation, 
management, and release of documentation.  

Going from the DevOps definition provided by Atlassian linked in the previous 
paragraph, then, we could loosely define DocOps as:

> "a set of practices that works to automate and integrate the process 

Reviewer: Can this be a blockquote instead of just a quoted paragraph?

Author: I actually couldn't figure out the directive to use for that with Sphinx. Any tips @ericholscher ?

Reviewer 2: Just an indent I think. https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#block-quotes

of developing documentation across engineering, product, support, and, of 
course, technical writing teams." 

Just like DevOps, DocOps also emphasizes collaboration and continuous 
improvement. Below, we've adapted the traditional DevOps lifecycle diagram to
represent DocOps:

.. image:: /_static/img/guide/docops-lifecycle.png

.. _DevOps: https://www.atlassian.com/devops

Who practices DocOps?
---------------------

To answer this question, it might be best to examine the responsibilities 
that a DocOps practitioner or team might have. 

* Selecting and providing support for tools used to produce documentation -- markup 
  language, static site generator (SSG), hosting provider, etc.
* Defining processes and established ways of working, including where docs fit
  into agile or other development methodologies.
* Defining documentation priorities and deliverables
* Providing training to authors to enable contribution to the docs
* Managing content     

> * Building and maintaining a theme for an SSG

Reviewer: This is a great list. As a DocOps practitioner, I also model product divergence (versioning) of all kinds and map it to the documentation infrastructure. The main reason teams hire me is to implement version management. And as long as we're mentioning theming, we're also usually maintaining build scripts to coordinate it all and make docs building as simple as possible on the commandline.

Basically, I tell writing teams that in the end I'm going to be their command-line concierge.

Author: ooh, excellent suggestions!

```suggestion
* Building and maintaining a theme for an SSG
* Modeling and implementing product versioning strategies as they pertain to documentation
* Creating and maintaining build scripts 
* Keeping the barrier to entry for making docs low to make contributing easy and (dare we say it?) fun
```

This is by no means an exhaustive list. If you have responsibilities that 
overlap with or incorporate some of these, then you may be a DocOps practioner.

It's also important to note that while DocOps tends to be associated with 

> docs-as-code, it's not a requirement. You might use DITA, a CMS, a wiki, etc.; 

Reviewer: Love this point!

if you're involved in the areas listed above, regardless of tool set, there's 
a good chance you might be DocOps!

DocOps Resources
----------------

If you're interested in DocOps, join the conversation in the Write the Docs Slack `#docops`_ channel!

> * `CA Technologies`_ introduced the DocOps approach to technical content

Reviewer: I would say they named it. This was already going on in other places when they released that article. That is the first known use of "DocOps", the term -- they also tried to trademark it (ugh) even though they weren't doing anything with it externally, and their DocOps website was _not on their own DocOps_ inrastructure, and they've done literally nothing noteworthy other than name it and I think give one talk. There's some risk here since CA is a proprietary operation that they will try to protect the name DocOps, so I would strongly caution against giving them much credit.

Author: Excellent insights, thanks!

```suggestion
* `CA Technologies`_ was among the first to publicly use the term "DocOps" to describe their approach to technical content
```

DocOps
======

author: Jodie Putrino & the Write the Docs community

What is DocOps, anyway?
-----------------------

> Along with the advent of `docs-as-code`_, the technical writing community 

Reviewer 3: ```suggestion
Along with the advent of :doc:`/guide/docs-as-code`, the technical writing community 
```

has seen the growth of a new specialization called "DocOps" (or "doc-ops").

But what does DocOps actually mean, and who actually "does" DocOps?

This subject came up for discussion during the WTD NA 2021 Unconference session. 
This page represents some of the ideas presented during that session.

Generally speaking, "DocOps" is like `DevOps`_. Instead of applying broadly to 
software development, though, DocOps specifically applies to the creation, 
management, and release of documentation.  

Going from the DevOps definition provided by Atlassian linked in the previous 
paragraph, then, we could loosely define DocOps as:

"a set of practices that works to automate and integrate the process 
of developing documentation across engineering, product, support, and, of 
course, technical writing teams." 

Just like DevOps, DocOps also emphasizes collaboration and continuous 
improvement. Below, we've adapted the traditional DevOps lifecycle diagram to
represent DocOps:

.. image:: /_static/img/guide/docops-lifecycle.png

> .. _docs-as-code: https://www.writethedocs.org/guide/docs-as-code/

Reviewer 3: ```suggestion
```
